prebid-server/pbs-modules/rules-engine.md

```diff
@@ -0,0 +1,805 @@
+                    "conditions": ["true","false","app","false","false"],
+                    "results": [{
+                        "function": "excludeBidders",
+                        "args": [{
+                            "bidders": ["bidderA", "bidderB"],
+                            "analyticsValue": "leaf3"
+                        }]
+                    }]
+                  },{
+                    "conditions": ["true","false","*","*","*"],
+                    "results": [{
+                        "function": "excludeBidders",
+                        "args": [{
+                            "bidders": ["bidderC"],
+                            "analyticsValue": "leaf4"
+                        }]
+                    }]
+                  },{
+                    "conditions": ["false","true","web","true","true"],
+                    "results": [{
+                        "function": "excludeBidders",
+                        "args": [{
+                            "bidders": ["bidderA", "bidderE"],
+                            "analyticsValue": "leaf5"
+                        }]
+                    }]
+                  },{
+                    "conditions": ["false","true","web","true","false"],
+                    "results": [{
+                        "function": "excludeBidders",
+                        "args": [{
+                            "bidders": ["bidderA", "bidderB","bidderE"],
+                            "analyticsValue": "leaf6"
+                        }]
+                    }]
+                  },{
+                    "conditions": ["false","true","web","false","*"],
+                    "results": [{
+                        "function": "excludeBidders",
+                        "args": [{
+                            "bidders": ["bidderA", "bidderD","bidderF"],
+                            "analyticsValue": "leaf7"
+                        }]
+                    }]
+                  },{
+                    "conditions": ["false","true","*","*","true"],
+                    "results": [{
+                        "function": "excludeBidders",
+                        "args": [{
+                            "bidders": ["bidderA", "bidderC"],
+                            "analyticsValue": "leaf8"
+                        }]
+                    }]
+                  },{
+                    "conditions": ["false","true","*","*","false"],
+                    "results": [{
+                        "function": "excludeBidders",
+                        "args": [{
+                            "bidders": ["bidderA", "bidderF"],
+                            "analyticsValue": "leaf9"
+                        }]
+                    }]
+                  },{
+                    "conditions": ["false","false","web","*","*"],
+                    "results": [{
+                        "function": "excludeBidders",
+                        "args": [{
+                            "bidders": [],
```

Comment:
As mentioned earlier, Go is considering it an error when bidders is empty. Couldn't you just omit this set of conditions in which case the default would be executed? I guess perhaps this has value in case you want to specify defaults but then not apply those defaults in certain instances like this. What do you think?